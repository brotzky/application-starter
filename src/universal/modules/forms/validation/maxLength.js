export default function(max) {
  if (!max) {
    return '';
  }

  return function(value) {
    return value.length <= max
      ? undefined
      : {
          defaultMessage: 'cannot be more than {max} characters',
          id: 'form.errors.maxLength',
          values: { max: max },
        };
  }
}
