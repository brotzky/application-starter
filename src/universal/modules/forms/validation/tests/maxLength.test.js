import maxLength from '../maxLength';

const checkIfOverThree = maxLength(3);

describe('[Form Validation] maxLength', () => {
  describe('maxLength()', () => {
    it('should return an empty string if nothing is passed it', () => {
      expect(maxLength()).toMatchSnapshot();
    });

    it('should return a function', () => {
      expect(maxLength(3)).toMatchSnapshot();
    });

    it('should return no errors if the number passed is less than the originally specified amount', () => {
      expect(checkIfOverThree('123')).toMatchSnapshot();
    });

    it('should return an error object when an amount over the specified is passed in', () => {
      expect(checkIfOverThree('1234')).toMatchSnapshot();
    });
  });
});
