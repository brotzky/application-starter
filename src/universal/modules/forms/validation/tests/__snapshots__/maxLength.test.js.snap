// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] maxLength maxLength() should return a function 1`] = `[Function]`;

exports[`[Form Validation] maxLength maxLength() should return an empty string if nothing is passed it 1`] = `""`;

exports[`[Form Validation] maxLength maxLength() should return an error object when an amount over the specified is passed in 1`] = `
Object {
  "defaultMessage": "cannot be more than {max} characters",
  "id": "form.errors.maxLength",
  "values": Object {
    "max": 3,
  },
}
`;

exports[`[Form Validation] maxLength maxLength() should return no errors if the number passed is less than the originally specified amount 1`] = `undefined`;
